### ISSUE TITLE
Possible problem with postgres bigint id's: Cannot return null for non-nullable field Invoices.invoices.

### ISSUE THREAD

**Issue by lastmjs** [aerogear/graphback#1423]
In my database, I have `Invoices` and `Invoice_line_items`. There is a one to many relationship from `Invoices` -> `Invoice_line_items`. `Invoice_line_items` has a field called `invoice_id`, of type bigint.

I'm not sure, but I think the bigint postgres type for the id has something to do with it, I'm seeing this problem with another relationship that uses a bigint for the foreign key id...not sure, but seems likely.

Here is my model:

```graphql
""" @model """
type Invoices {
    id: ID!

    """
    @oneToMany field: 'invoice', key: 'invoice_id'
    """
    invoices: [Invoice_line_items!]!
}

""" @model """
type Invoice_line_items {
    id: ID!
    invoice: Invoices!
}
```

When I try to do this query:

```graphql
query {
  findInvoices {
    items {
      id
      invoices {
        id
      }
    }
  }
}
```
I get this error: `Cannot return null for non-nullable field Invoices.invoices.`

But when I do it the other way around, the invoices are properly associated with the invoice line items:

```graphql
query {
  findInvoice_line_items {
    items {
      id
      invoice {
        id
      }
    }
  }
}
```
Gives me:

```json
{
  "data": {
    "findInvoice_line_items": {
      "items": [
        {
          "id": "1",
          "invoice": {
            "id": "1"
          }
        },
        {
          "id": "2",
          "invoice": {
            "id": "1"
          }
        },
        {
          "id": "3",
          "invoice": {
            "id": "2"
          }
        },
        {
          "id": "4",
          "invoice": {
            "id": "2"
          }
        },
        {
          "id": "5",
          "invoice": {
            "id": "3"
          }
        },
        {
          "id": "6",
          "invoice": {
            "id": "3"
          }
        },
        {
          "id": "7",
          "invoice": {
            "id": "4"
          }
        },
        {
          "id": "8",
          "invoice": {
            "id": "4"
          }
        },
        {
          "id": "9",
          "invoice": {
            "id": "5"
          }
        },
        {
          "id": "10",
          "invoice": {
            "id": "5"
          }
        },
        {
          "id": "11",
          "invoice": {
            "id": "6"
          }
        },
        {
          "id": "12",
          "invoice": {
            "id": "6"
          }
        },
        {
          "id": "13",
          "invoice": {
            "id": "7"
          }
        },
        {
          "id": "14",
          "invoice": {
            "id": "7"
          }
        },
        {
          "id": "15",
          "invoice": {
            "id": "8"
          }
        },
        {
          "id": "16",
          "invoice": {
            "id": "8"
          }
        },
        {
          "id": "17",
          "invoice": {
            "id": "9"
          }
        },
        {
          "id": "18",
          "invoice": {
            "id": "9"
          }
        },
        {
          "id": "19",
          "invoice": {
            "id": "10"
          }
        },
        {
          "id": "20",
          "invoice": {
            "id": "10"
          }
        }
      ]
    }
  }
}
```
I've messed around the the relationship annotations, and I can't get it to work.

**Comment by lastmjs** [comment]
I'm seeing this problem with another relationship as well. My companies have invoices, and I can select the invoices from a companies query, but the invoices cannot select the companies.

**Comment by craicoverflow** [comment]
Hey @lastmjs! I've tried to reproduce this issue but was not able to. This could be because I am using `graphql-migrations` to set up my database, whereas yours does not use this. I'll show you my two table structures, maybe you can share yours so we can see if there are any differences:

**Invoices:**

```
                        Table "public.invoices"
 Column |  Type  |                       Modifiers                       
--------+--------+-------------------------------------------------------
 id     | bigint | not null default nextval('invoices_id_seq'::regclass)
Indexes:
    "invoices_pkey" PRIMARY KEY, btree (id)
Referenced by:
    TABLE "invoice_line_items" CONSTRAINT "invoice_line_items_invoice_id_foreign" FOREIGN KEY (invoice_id) REFERENCES invoices(id)
```

**Invoice_line_items:**

```
                           Table "public.invoice_line_items"
   Column   |  Type  |                            Modifiers                            
------------+--------+-----------------------------------------------------------------
 invoice_id | bigint | not null
 id         | bigint | not null default nextval('invoice_line_items_id_seq'::regclass)
Indexes:
    "invoice_line_items_pkey" PRIMARY KEY, btree (id)
Foreign-key constraints:
    "invoice_line_items_invoice_id_foreign" FOREIGN KEY (invoice_id) REFERENCES invoices(id)
```

Which version of Graphback are you using?

**Comment by lastmjs** [comment]
Thanks for helping me out.

I am using `graphback@0.14.0-alpha1` and `PostgreSQL 12.1`.

I am currently using Active Record to create and migrate my database, as we're refactoring a Rails application.

Here's the description for my invoices table:

```
                                   Table "public.invoices"
   Column   |       Type        | Collation | Nullable |               Default                
------------+-------------------+-----------+----------+--------------------------------------
 id         | bigint            |           | not null | nextval('invoices_id_seq'::regclass)
 company_id | bigint            |           |          | 
 due_date   | date              |           |          | 
 status     | integer           |           | not null | 0
 for_month  | character varying |           |          | 
 stripe_id  | character varying |           |          | 
Indexes:
    "invoices_pkey" PRIMARY KEY, btree (id)
    "index_invoices_on_company_id" btree (company_id)
Foreign-key constraints:
    "fk_rails_433d4c368d" FOREIGN KEY (company_id) REFERENCES companies(id)
Referenced by:
    TABLE "invoice_line_items" CONSTRAINT "fk_rails_80427eb9d3" FOREIGN KEY (invoice_id) REFERENCES invoices(id)
```
And for invoice_line_items:

```
                                    Table "public.invoice_line_items"
   Column    |       Type        | Collation | Nullable |                    Default                     
-------------+-------------------+-----------+----------+------------------------------------------------
 id          | bigint            |           | not null | nextval('invoice_line_items_id_seq'::regclass)
 invoice_id  | bigint            |           |          | 
 description | character varying |           |          | 
 amount      | integer           |           |          | 
Indexes:
    "invoice_line_items_pkey" PRIMARY KEY, btree (id)
    "index_invoice_line_items_on_invoice_id" btree (invoice_id)
Foreign-key constraints:
    "fk_rails_80427eb9d3" FOREIGN KEY (invoice_id) REFERENCES invoices(id)
```

**Comment by craicoverflow** [comment]
Thanks for the table information. I've still been unable to reproduce this with those version you have specified. It may be an incompatibility with some other dependency in your server? Here are a few more of my dependencies and their versions that may be relevant:

`apollo-server-express`: 2.14.3
`graphback`: 0.14.0-alpha1
`@graphback/runtime-knex`: 0.14.0-alpha1
`graphql`: 14.6.0
`postgres`: 12.1 (Docker container)

![image](https://user-images.githubusercontent.com/11743717/84640516-b4175200-aef1-11ea-8d91-d3841be6b8eb.png)

My query logs:

```
querying object Invoice_line_items with filter undefined
Logger.ts:21
fetching single object Invoices with args {"id":"1"}
Logger.ts:21
querying object Invoice_line_items with filter undefined
Logger.ts:21
fetching single object Invoices with args {"id":"1"}
Logger.ts:21
querying object Invoice_line_items with filter undefined
Logger.ts:21
fetching single object Invoices with args {"id":"1"}
Logger.ts:21
querying object Invoice_line_items with filter undefined
Logger.ts:21
fetching single object Invoices with args {"id":"1"}
Logger.ts:21
querying object Invoice_line_items with filter undefined
Logger.ts:21
fetching single object Invoices with args {"id":"1"}
Logger.ts:21
querying object Invoice_line_items with filter undefined
```

If you were to temporarily remove change `[Invoice_line_items!]!` to `invoices: [Invoice_line_items]` does it return null or the list of items?

**Comment by lastmjs** [comment]
Somehow, the issue seems to be resolved for me now. I can do all of the problematic queries that I couldn't before. I believe that I've recreated my database a couple times since I posted this, perhaps there was a problem with the structure before, not sure

**Comment by wtrocki** [comment]
We have released alpha4  with latest fixes

**Comment by lastmjs** [comment]
This is actually not resolved. I believe I know how to reproduce. Use postgres 9.6.11. On my local machine, the following query with the table structures I gave you above gives me the error:

```graphql
query {
  findInvoice_line_items {
    items {
      id
      invoice {
        id
        invoice_line_items {
          id
        }
      }
    }
  }
}
```
If I switch to a docker container with postgres 12, the error goes away. I found this out because my staging environment is on postgres 9.6.12, and it has the error, while I switched apparently to postgres 12 locally after I opened this issue. I'd love to know if you can reproduce it and see if we can find the root of the issue.

**Comment by wtrocki** [comment]
CC @machi1990 let's chat about this in the morning

**Comment by lastmjs** [comment]
Testing on postgres 11.1, and the issue remains

**Comment by lastmjs** [comment]
Now all of the postgres versions that I'm trying, on AWS RDS and locally on docker, are broken with the same error.

**Comment by lastmjs** [comment]
> If you were to temporarily remove change [Invoice_line_items!]! to invoices: [Invoice_line_items] does it return null or the list of items?

@craicoverflow When I change `invoice_line_items: [Invoice_line_items!]!` to `invoice_line_items: [Invoice_line_items]`, all of the invoice_line_items in the result for the following query or null:

```graphql
query {
  findInvoice_line_items {
    items {
      id
      invoice {
        id
        invoice_line_items {
          id
        }
      }
    }
  }
}
```
Result:
```json
{
  "data": {
    "findInvoice_line_items": {
      "items": [
        {
          "id": "1",
          "invoice": {
            "id": "1",
            "invoice_line_items": null
          }
        },
        {
          "id": "2",
          "invoice": {
            "id": "1",
            "invoice_line_items": null
          }
        },
        {
          "id": "3",
          "invoice": {
            "id": "2",
            "invoice_line_items": null
          }
        },
        {
          "id": "4",
          "invoice": {
            "id": "2",
            "invoice_line_items": null
          }
        },
        {
          "id": "5",
          "invoice": {
            "id": "3",
            "invoice_line_items": null
          }
        },
        {
          "id": "6",
          "invoice": {
            "id": "3",
            "invoice_line_items": null
          }
        },
        {
          "id": "7",
          "invoice": {
            "id": "4",
            "invoice_line_items": null
          }
        },
        {
          "id": "8",
          "invoice": {
            "id": "4",
            "invoice_line_items": null
          }
        },
        {
          "id": "9",
          "invoice": {
            "id": "5",
            "invoice_line_items": null
          }
        },
        {
          "id": "10",
          "invoice": {
            "id": "5",
            "invoice_line_items": null
          }
        },
        {
          "id": "11",
          "invoice": {
            "id": "6",
            "invoice_line_items": null
          }
        },
        {
          "id": "12",
          "invoice": {
            "id": "6",
            "invoice_line_items": null
          }
        },
        {
          "id": "13",
          "invoice": {
            "id": "7",
            "invoice_line_items": null
          }
        },
        {
          "id": "14",
          "invoice": {
            "id": "7",
            "invoice_line_items": null
          }
        },
        {
          "id": "15",
          "invoice": {
            "id": "8",
            "invoice_line_items": null
          }
        },
        {
          "id": "16",
          "invoice": {
            "id": "8",
            "invoice_line_items": null
          }
        },
        {
          "id": "17",
          "invoice": {
            "id": "9",
            "invoice_line_items": null
          }
        },
        {
          "id": "18",
          "invoice": {
            "id": "9",
            "invoice_line_items": null
          }
        },
        {
          "id": "19",
          "invoice": {
            "id": "10",
            "invoice_line_items": null
          }
        },
        {
          "id": "20",
          "invoice": {
            "id": "10",
            "invoice_line_items": null
          }
        }
      ]
    }
  }
}
```

**Comment by lastmjs** [comment]
I have found the root of the problem, and it has nothing to do with the postgres DB version. In this issue https://github.com/aerogear/graphback/issues/1432 I mentioned creating a custom type and resolver for `stripe_invoice`. Well, in the resolvers object, I had overriden `Invoices` like this:

Simplified example:
```typescript
resolvers: {
   ...resolvers,
   Invoices: {
    stripe_invoice: stripeInvoice
   }
}
```
So the `Invoices` were resolved fine for primitives, but for any relations I had essentially removed those resolvers. By doing this the issue is fixed:

```typescript
resolvers: {
   ...resolvers,
   Invoices: {
    ...resolvers.Invoices,
    stripe_invoice: stripeInvoice
   }
}
```
